include/sparrow/layout/primitive_layout/primitive_data_access.hpp

```diff
@@ -278,5 +355,175 @@ namespace sparrow
         {
             return *p_proxy;
         }
+
+        /**********************************************
+         * primitive_data_access<bool> implementation *
+         **********************************************/
+
+        inline primitive_data_access<bool>::primitive_data_access(arrow_proxy& proxy, size_t data_buffer_index)
+            : p_proxy(&proxy)
+            , m_data_buffer_index(data_buffer_index)
+            , m_view(get_data_view())
+            , m_dummy_buffer()
+            , m_adaptor(&m_dummy_buffer, 0u)
+            , m_empty_adaptor(true)
+        {
+            if (p_proxy->is_created_with_sparrow())
+            {
+                m_adaptor = get_data_adaptor();
+                m_empty_adaptor = false;
+            }
```

Comment:
Can you set the m_adaptor and m_empty_adaptor thanks static functions

bitset_adaptor bitset_adaptor_init(const arrow_proxy& proxy)
{
if (p_proxy->is_created_with_sparrow())
            {